### PR TITLE
Add `init_optimizers` test for predict

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Just updated a minor test to ensure that optimizers/schedulers are not required during `trainer.predict`.

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by ananthsub — APPROVED** [review]

File: tests/trainer/optimization/test_optimizers.py [hunk 1]
```diff
@@ -299,9 +299,9 @@ def configure_optimizers(self):
     assert trainer.lr_schedulers[1]["scheduler"]._step_count == expected_steps[1]
 
 
-@pytest.mark.parametrize("fn", ("validate", "test"))
-def test_init_optimizers_during_evaluation(tmpdir, fn):
-    """Test that optimizers is an empty list during evaluation."""
+@pytest.mark.parametrize("fn", ("validate", "test", "predict"))
+def test_init_optimizers_during_evaluation_and_predict(tmpdir, fn):
```
Comment (awaelchli) [inline tests/trainer/optimization/test_optimizers.py:303]:
```suggestion
def test_init_optimizers_during_evaluation_and_prediction(tmpdir, fn):
```

**Review by awaelchli — APPROVED** [review]

File: tests/trainer/optimization/test_optimizers.py [hunk 2]
```diff
@@ -299,9 +299,9 @@ def configure_optimizers(self):
...
+    """Test that optimizers is an empty list during evaluation and predict."""
```
Comment (awaelchli) [inline tests/trainer/optimization/test_optimizers.py:304]:
```suggestion
    """Test that optimizers is an empty list during evaluation and prediction."""
```
grammar police on duty :))))

**Review by tchaton — APPROVED** [review]
LGTM !
